backend/src/neuroagent/tools/obi_expert.py

```
@@ -0,0 +1,666 @@
+        if isinstance(flattened, str):
+            return flattened
+
+        # If it's still a list, join all text elements together
+        if isinstance(flattened, list):
+            text_parts = []
+            for item in flattened:
+                if isinstance(item, str):
+                    text_parts.append(item)
+                elif isinstance(item, dict):
+                    # Extract text from various block types
+                    if "title" in item:
+                        text_parts.append(item["title"])
+                    elif "text" in item:
+                        text_parts.append(item["text"])
+                    elif "content" in item and isinstance(item["content"], str):
+                        text_parts.append(item["content"])
+
+            return " ".join(text_parts) if text_parts else ""
+
+        # Fallback: convert to string
+        return str(flattened) if flattened else ""
+
+
+class GlossaryItemDocument(SanityDocument):
+    """Schema for glossary item documents."""
+
+    name: str
+    description: str
+    definition: str | None
+
+    sanity_mapping: ClassVar[dict[str, str]] = {
+        **SanityDocument.sanity_mapping,
+        "name": "Name",
+        "description": "Description",
+        "definition": "definition",
+    }
+
+    @field_validator("definition", mode="before")
+    @classmethod
+    def flatten_definition(cls, v: list[dict[str, Any]] | None) -> str | None:
+        """Flatten the definition from portable text blocks into a single string."""
+        if v is None:
+            return None
+
+        # Use the existing flatten_portable_text function to process the definition
+        flattened = flatten_portable_text(v)
+
+        # The flatten_portable_text function should return a string for portable text
+        if isinstance(flattened, str):
+            return flattened
+
+        # Fallback: convert to string
+        return str(flattened) if flattened else ""
```

In theory `flatten_portable_text` should return a string if the block is not `None` right ? Shouldn't we instead raise an error if it is not a string so that we can catch potential bugs ?